workspace_status.sh

```diff
@@ -11,10 +11,16 @@ echo "GIT_BRANCH $git_branch"
 git_tree_status=$(git diff-index --quiet HEAD -- && echo 'Clean' || echo 'Modified')
 echo "GIT_TREE_STATUS $git_tree_status"
 
-exact=$(git describe --exact-match 2>/dev/null | sed -e s/^v// | cut -d - -f 1 || echo 2.11.0)
+exact=$(git describe --exact-match 2>/dev/null | sed -e s/^v// | cut -d - -f 1 || echo 2.12.0)
 echo "STABLE_APP_VERSION $exact"
 
-additional_version=$(git describe --exact-match 2>/dev/null | cut -d - -f 2- || echo "pre.$(git describe --long --dirty=".$(git diff HEAD | sha256sum | cut -c 1-10)" | rev | cut -d - -f 1 | rev)")
+SHA256SUM="sha256sum"
+if command -v shasum &> /dev/null; then
```

Comment:
This logic is ‘use `sha256sum`, but is `shasum` exists use it instead’; shouldn’t the logic be ‘use sha256sum if it exists; use `shasum` if it exists; otherwise error’?

```sh
if command -v sha256sum &> /dev/null; then
  SHA256SUM="sha256sum"
else
  if command -v shasum &> /dev/null; then
    SHA256SUM="shasum"
  else
     echo "neither sha256sum nor shasum in PATH" >&2
     exit 1
  fi
fi
```

Reply:
We don't need to error.  This has been broken on Mac OS for a while and the lack of erroring hasn't been harmful.  

What this code does is set AdditionalVersion in your pachd/pachctl binary to the git hash, plus a hash of your diffs if there are any.  That way, you always have a unique version number.  But if we can't hash your diffs, then you can just reuse the same version number.  It's only really used when comparing pachd versions in pachdev push... if the version number of the new code equals the old code, then we may return a little early.  But it doesn't interfere with pushing or anything, which is based on the sha256 of the built containers.

Reply:
Still seems odd to me, but not a huge deal I guess.